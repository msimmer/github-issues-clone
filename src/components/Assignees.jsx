// @flow

import * as React from "react";
import Assignee from "./Assignee";

const Assignees = (props: { edges: Object[] }): React.Node => {
  if (!props.edges || props.edges.length < 1) return null;
  return props.edges.map(({ node }, i) => (
    <Assignee key={node.id} index={i} {...node} />
  ));
};

export default Assignees;
