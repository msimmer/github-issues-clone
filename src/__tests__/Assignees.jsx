import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Assignees from "../components/Assignees";

configure({ adapter: new Adapter() });

describe("<Assignees />", () => {
  const edges = [
    {
      id: 1,
      login: "foo",
      avatarUrl: "foo.jpg"
    },
    {
      id: 2,
      login: "bar",
      avatarUrl: "bar.jpg"
    }
  ];

  it("matches the snapshots", () => {
    const component = shallow(<Assignees edges={[]} />);
    expect(toJSON(component)).toMatchSnapshot();

    component.setProps({
      edges: edges.map((a, index) => ({ node: { ...a, index } }))
    });
    expect(toJSON(component)).toMatchSnapshot();
  });
});
