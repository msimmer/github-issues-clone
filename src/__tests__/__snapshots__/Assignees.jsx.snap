// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Assignees /> matches the snapshots 1`] = `""`;

exports[`<Assignees /> matches the snapshots 2`] = `
Array [
  <Assignee
    avatarUrl="foo.jpg"
    id={1}
    index={0}
    key="1"
    login="foo"
  />,
  <Assignee
    avatarUrl="bar.jpg"
    id={2}
    index={1}
    key="2"
    login="bar"
  />,
]
`;
